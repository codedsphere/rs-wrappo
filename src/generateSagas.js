import {all, takeLatest, fork} from 'redux-saga/effects';
import map from 'lodash-es/map';

function* generateSagaInd({sagas}) {
  return yield fork(function* root() {
    const sagasList = [];
    if (sagas) {
      for (const [key] of Object.entries(sagas)) {
        sagasList.push(yield takeLatest(key, sagas[key]));
      }
    }
    return yield all(sagasList);
  });
}

export default function generateSagas(sagasList) {
  const fullSagaList = [];

  map(sagasList, saga => {
    fullSagaList.push(generateSagaInd(saga));
  });
  return function* root() {
    yield all(fullSagaList);
  };
}
