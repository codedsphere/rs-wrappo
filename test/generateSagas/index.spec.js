import generateSagas from 'generateSagas';
import exampleSaga from '../__mocks__/rootSaga/exampleSaga';

describe('generateSagas', () => {
  it('should have sagas', () => {
    const sagas = generateSagas([exampleSaga]);
    expect(sagas).toMatchSnapshot();
  });
});
