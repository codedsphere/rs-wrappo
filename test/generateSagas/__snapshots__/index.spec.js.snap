// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSagas should have sagas 1`] = `[Function]`;
